commons/ava_networks/high_level_gecko_client.go

```diff
@@ -40,6 +43,46 @@ func NewGeckoUser(username string, password string) *GeckoUser {
 }
 
 
+func (highLevelGeckoClient HighLevelGeckoClient) AddValidatorOnSubnet(
+	nodeId string,
+	pchainAddress string,
+	stakeAmount int64) error {
+	client := highLevelGeckoClient.client
+	currentPayerNonce, err := highLevelGeckoClient.getCurrentPayerNonce(pchainAddress)
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to get payer nonce from address %s", pchainAddress)
+	}
+	stakingStartTime := time.Now().Add(TIME_UNTIL_STAKING_BEGINS).Unix()
+	addStakerUnsignedTxn, err := client.PChainApi().AddDefaultSubnetValidator(
+		nodeId,
+		stakingStartTime,
+		time.Now().Add(TIME_UNTIL_STAKING_ENDS).Unix(),
+		stakeAmount,
+		currentPayerNonce + 1,
```

Comment:
Do we need to store this incremented value anywhere?

Reply:
@mieubrisse I'm taking the strategy of checking the blockchain before each call in order to get the current nonce, rather than storing it